backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -234,6 +234,12 @@ public Building Update(Building building)
             // Only administrators can dispose a property.
             if (building.ClassificationId == 4 && !isAdmin) throw new NotAuthorizedException("Building classification cannot be changed to disposed.");
 
+            // Only administrators can set a building to demolished
+            if(building.ClassificationId == 5 && isAdmin) throw new NotAuthorizedException("Building classification cannot be changed to demolished.");
```

Comment:
Thanks for adding back the enum, but could you apply it here as well? sorry to be a pain.

Reply:
No problem at all, forgot about this area. Thanks for catching it!